xgi/generators/classic.py

```diff
@@ -154,3 +157,59 @@ def trivial_hypergraph(n=1, create_using=None, default=None):
     H.add_nodes_from(nodes)
 
     return H
+
+
+def complete_hypergraph(N, order=None, max_order=None, include_singletons=False):
+    """
+    Generate a complete hypergraph, i.e. one that contains all possible hyperdges
+    at a given `order` or up to a `max_order`.
+
+    Parameters
+    ----------
+
+    N : int
+        Number of nodes
+    order : int or None
+        If not None (default), specifies the single order for which to generate hyperedges
+    max_order : int or None
+        If not None (default), specifies the maximum order for which to generate hyperedges
+    include_singletons : bool
+        Whether to include singleton edges (default: False). This argument is discarded
+        if max_order is None.
+
+    Return
+    ------
+    Hypergraph object
+        A complete hypergraph with `N` nodes
+
+    Note
+    ----
+    Only one of `order` and `max_order` can be specified by and int (not None).
```

Comment:
I wonder if you could comment on the maximum edge order that is computationally feasible. Should there be a more conservative default so that people don't try to compute the complete hypergraph for $N=100$?

Reply:
You mean mention that max_order < N -1?

The default (None) does not compute anything. So it's not computing all possible orders by default.

Reply:
I see - I should have looked more closely. Nonetheless, maybe you could add a rough estimate in the Notes section on max order/max $N$ for which computation is feasible. Something along the lines of "When the size of the hypergraph is smaller than 20, there are no restrictions on the max order. For all hypergraphs larger than this, it is advisable to set max_order to be less than 5 to make the computation feasible." It probably is more complicated than that, but just to warn the user that, for example, a hypergraph with 1,000 nodes and max_order=1000 will not work.